beangulp/testing.py

```diff
@@ -76,90 +121,72 @@ def log(msg, level=0, **kwargs):
     desired output.  The desired output is stored in Beancount ledger
     files with an header containing additional metadata.  These files
     are named as the SHA1 digest of the documents and can be generated
     with the --generate option.
 
-    DOCUMENTS can be files or directories.  Directories are walked and
-    the importer is called on all files.  When no DOCUMENTS path is
-    specified on the command line, the tests/documents/ directory
-    relative to the test script location is used.  All documents for
-    which an expected output file exist must be positively identified
-    by the importer.
-
-    Expected output files are searched in the tests/expected/$NAME/
-    directory relative to the test script location, where $NAME is
-    replaced with the importer name, if the path exists, or in the
-    tests/expected/ directory.  This path can be overwritten through
-    the --expected DIR option.
-
+    DOCUMENTS can be files or directories. Directories are walked and the
+    importer is called on all files found, except for .beancount files.
     """
     assert len(ctx.importers) == 1
     importer = ctx.importers[0]
 
-    import __main__
-    root = os.path.dirname(__main__.__file__)
-
-    if not documents:
-        documents = (os.path.join(root, 'tests','documents'), )
-
-    if not expected:
-        expected = os.path.join(root, 'tests', 'expected', importer.name())
-        if not os.path.exists(expected):
-            expected = os.path.join(root, 'tests', 'expected')
-
     verbosity = verbose - quiet
     log = logger(verbosity)
     failures = 0
 
-    for doc in walk(documents):
-        # unless verbose mode is enabled, do not output a newline so
+    for doc in walk(documents, r".*\.beancount$"):
+        # Unless verbose mode is enabled, do not output a newline so
         # the test result is printed on the same line as the test
         # document filename.
-        log(f'∙ {os.path.relpath(doc):}', nl=verbosity > 0)
-
-        # compute the path to the expected output file
-        path = os.path.join(expected, sha1sum(doc) + '.beancount')
-        f = cache.get_file(os.path.abspath(doc))
-
-        if importer.identify(f):
-            account, date, name, entries = run_importer(importer, f)
-            log(f'  {os.path.relpath(path):}', 1)
-            log(f'  {account.replace(":", "/"):}/{date:%Y-%m-%d}-{name:}', 1)
+        log(f'* {doc}', nl=verbosity > 0)
+
+        # Compute the path to the expected output file.
+        expected_filename = "{}.beancount".format(doc)
+        cached_file = cache.get_file(path.abspath(doc))
+
+        # Run the importer's identify() method.
+        if importer.identify(cached_file):
+            account, date, name, entries = run_importer(importer, cached_file)
+            log(f'  {cached_file}', 1)
+            if account and date and name:
+                # TODO(blais): Figure out how to produce a log entry when some
+                # of these are None.
```

Comment:
I thought that returning None for any of these is a bug in the importer. Isn't it?

Reply:
I think file_date() and file_name() were always optional. 
See the docstrings here:
https://github.com/beancount/beangulp/blob/master/beangulp/importer.py#L84

I'll update the types in my PR so that account isn't optional.

Sometimes there's just no date... what does one do?
Also, if the filename doesn't stand to get renamed, it's okay to return None and keep the name by default.
Let's keep these optional, to ease the burden on users.

It's high time we add types to these :-)
(Will update PR)